ui/src/app/kvm/components/StorageColumn/StoragePopover/StoragePopover.tsx

```diff
@@ -4,20 +4,24 @@ import { Fragment } from "react";
 import Meter from "app/base/components/Meter";
 import Popover from "app/base/components/Popover";
 import { COLOURS } from "app/base/constants";
-import type {
-  KVMStoragePoolResource,
-  KVMStoragePoolResources,
-} from "app/kvm/types";
+import type { KVMStoragePoolResources } from "app/kvm/types";
+import { getSortedPoolsArray } from "app/kvm/utils";
+import type { Pod } from "app/store/pod/types";
 import { formatBytes } from "app/utils";
 
 type Props = {
   children: ReactNode;
+  defaultPoolId?: Pod["default_storage_pool"];
   pools: KVMStoragePoolResources;
 };
 
-const StoragePopover = ({ children, pools }: Props): JSX.Element => {
-  const poolsArray = Object.entries<KVMStoragePoolResource>(pools);
-  const showOthers = poolsArray.some((pool) => pool[1].allocated_other !== 0);
+const StoragePopover = ({
+  children,
+  defaultPoolId,
+  pools,
+}: Props): JSX.Element => {
+  const sortedPools = getSortedPoolsArray(pools, defaultPoolId);
+  const showOthers = sortedPools.some((pool) => pool[1].allocated_other !== 0);
```

Comment:
It might be clearer to destructure the pool rather than using the index, which can be a bit obtuse:

```suggestion
  const showOthers = sortedPools.some(([, pool]) => pool.allocated_other !== 0);
```